jmslib/src/test/kotlin/com/redhat/mqe/lib/InteractionTest.kt

```diff
@@ -0,0 +1,66 @@
+/*
+ * Copyright (c) 2018 Red Hat, Inc.
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.redhat.mqe.lib
+
+import com.redhat.mqe.lib.Main.main
+import org.junit.jupiter.api.Test
+import org.mockito.BDDMockito.given
+import org.mockito.Mockito.*
+import javax.jms.*
+
+class InteractionTest {
+    @Test
+    fun `test run sender no params`() {
+        val producer = mock(MessageProducer::class.java)
```

Comment:
Absolutely, I think the `producer` and so on should be public fields on a class, for convenience, so that one can create generic interlinked set of mocks by creating a class, then adjust them, then run a test.

I want to do that later, while creating the same test for a receiver and connector client. When the abstraction will be forced on me, because that is only way to avoid horrendous duplication.

Reply:
So, will do, but not just yet.